### PR TITLE
Testing - Use Argo v2.3.0

### PR DESCRIPTION
**Pull request by Ark-kun**
Verifying that all the system works with the final Argo v2.3.0 version.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1368)

### CODE REVIEW

File: test/deploy-kubeflow.sh [hunk 1]
```diff
@@ -56,4 +56,11 @@ ${KUBEFLOW_SRC}/scripts/kfctl.sh apply platform
 ${KUBEFLOW_SRC}/scripts/kfctl.sh generate k8s
 ${KUBEFLOW_SRC}/scripts/kfctl.sh apply k8s
 
+pushd ks_app
```
Comment (IronPan) [inline test/deploy-kubeflow.sh]: should we instead update the version in kfctl instead?
the change here will make the test environment different from production environment so test pass here might not mean it's working.

Comment (Ark-kun) [reply test/deploy-kubeflow.sh]: Done https://github.com/kubeflow/kubeflow/pull/3335

Comment (IronPan) [reply test/deploy-kubeflow.sh]: thanks. I think this PR is ready to go once this file is reverted.

**Comment by Ark-kun** [comment]
The tests succeed. I've upgraded Argo jsonnet in the kubeflow repo.

**Comment by eterna2** [comment]
Hey, just to update u, I have also updated `argo-models` to `v2.3.0`. 

Don't think there is any material change (w respect to the kfp sdk, argo `sidecar` -> `user_container`). 

But `kfp.dsl.Sidecar` is not based on the argo `sidecar` (it inherits from k8s container instead).

So u can probably up the version for `argo-models` for setup.py and requirement.txt, once u upgrades to argo 2.3.

**Comment by Ark-kun** [comment]
I've already upgraded Argo to 2.3

**Pull request closed without merging**